stdlib/pervasives.ml

```diff
@@ -369,7 +373,7 @@ external seek_out : out_channel -> int -> unit = "caml_ml_seek_out"
 external pos_out : out_channel -> int = "caml_ml_pos_out"
 external out_channel_length : out_channel -> int = "caml_ml_channel_size"
 external close_out_channel : out_channel -> unit = "caml_ml_close_channel"
-let close_out oc = flush oc; close_out_channel oc
+let close_out oc = (try flush oc with _ -> ()); close_out_channel oc
```

Comment:
@hcarty Note that this patch only makes `stdlib` and `threads` consistent.  In this case I chose the semantics as already implemented in `threads`, which I considered saner.  It will close the channel for sure, otherwise we would be leaking file descriptors.  If a user absolutely wants to make sure that their data has been flushed, they need to explicitly call `flush` themselves before calling `close_out`, thus potentially raising `Sys_error` exceptions.  `flush_all` also suppresses such exceptions, because the user could not possibly know anyway which channel was broken.